tests/test_support.bash

```diff
@@ -7,27 +7,34 @@ bats_require_minimum_version 1.5.0
 # To do so a test file may redefine `setup_file' and call `common_setup' before
 # writing their extensions.
 common_setup() {
-  if [[ -z "${FLOX_CLI:-}" ]]; then
-    if [[ -L ./result ]]; then
-      FLOX_PACKAGE="$(readlink ./result)"
-    else
-      FLOX_PACKAGE="$(flox build -A flox --print-out-paths --substituters "")"
-    fi
-    export FLOX_PACKAGE
-    export FLOX_CLI="$FLOX_PACKAGE/bin/flox"
-    FLOX_PACKAGE_FIRST8="$(
-      echo $FLOX_PACKAGE | dd bs=c skip=11 count=8 2>/dev/null
-    )"
-    export FLOX_PACKAGE_FIRST8
+  if ! command -v expect >/dev/null 2>&1; then
+    echo "ERROR: expect library needs to be in PATH."
+    return 1 
   fi
+
+  if [[ -z $FLOX_CLI ]]; then
+    echo "ERROR: FLOX_CLI (a path to the binary) needs to be declared."
+    return 1 
+  fi
+
+  export FLOX_PACKAGE="$(dirname $(dirname $FLOX_CLI))"
+  if [[ $FLOX_PACKAGE != /nix/store/* ]]; then
+    echo "ERROR: FLOX_PACKAGE (a path to the nix package in /nix/store) needs to be a /nix/store entry, but it is:\n\n  $FLOX_PACKAGE."
+    return 1 
+  fi
+
+  export FLOX_PACKAGE_FIRST8="$(echo $FLOX_PACKAGE | dd bs=c skip=11 count=8 2>/dev/null)"
+
   export TEST_ENVIRONMENT=_testing_
+
   # Remove any vestiges of previous test runs.
   $FLOX_CLI destroy -e "$TEST_ENVIRONMENT" --origin -f || :
-  NIX_SYSTEM="$(
+
+  export NIX_SYSTEM="$(
     $FLOX_CLI nix --extra-experimental-features nix-command show-config  \
       |awk '/system = / {print $NF}'
   )"
-  export NIX_SYSTEM
+
```

Comment:
```suggestion
export NIX_SYSTEM="$(nix eval --impure --expr builtins.currentSystem)"
```

Reply:
I can't commit the suggestion, but I'll manually fix it. Tnx.